packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -912,6 +922,15 @@ class _ModalScopeStatus extends InheritedWidget {
     description.add(FlagProperty('canPop', value: canPop, ifTrue: 'can pop'));
     description.add(FlagProperty('impliesAppBarDismissal', value: impliesAppBarDismissal, ifTrue: 'implies app bar dismissal'));
   }
+
+  @override
+  bool updateShouldNotifyDependent(covariant _ModalScopeStatus oldWidget, Set<Object> dependencies) {
```

Comment:
I believe you can use the more specific type here:

```suggestion
  bool updateShouldNotifyDependent(covariant _ModalScopeStatus oldWidget, Set<_ModalRouteAspect> dependencies) {
```

... and then you shouldn't need the `is _ModalRouteAspect` check below.